app/src/main/java/ch/epfl/cs311/wanderwave/model/spotify/SpotifyController.kt

```diff
@@ -99,6 +101,87 @@ class SpotifyController(private val context: Context) {
     }
   }
 
+  /**
+   * Get all the playlist, title, ... from spotify from the home page of the user.
+   *
+   * @return a Flow of ListItem which has all the playlist, title, ... from the home page of the
+   *   user.
+   * @author Menzo Bouaissi
+   * @since 2.0
+   * @last update 2.0
+   */
+  fun getAllElementFromSpotify(): Flow<List<ListItem>> {
+    val list: MutableList<ListItem> = emptyList<ListItem>().toMutableList()
+    return callbackFlow {
+      val callResult =
+          appRemote?.let { it ->
+            it.contentApi
+                .getRecommendedContentItems(ContentApi.ContentType.DEFAULT)
+                .setResultCallback {
+                  for (i in it.items) list += i
+                  trySend(list)
+                }
+                .setErrorCallback { trySend(list + ListItem("", "", null, "", "", false, false)) }
```

Comment:
Is this really what we want? To send empty tracks? Why not just not pushing anything?

Reply:
You can't return null instead of a flow, so I guess it's the best way to do it